lib/liquid/standardfilters.rb

```diff
@@ -194,8 +194,12 @@ def map(input, property)
         if property == "to_liquid".freeze
           e
         elsif e.respond_to?(:[])
-          r = e[property]
-          r.is_a?(Proc) ? r.call : r
+          begin
+            r = e[property]
+            r.is_a?(Proc) ? r.call : r
+          rescue TypeError
+            nil
```

Comment:
I don't think we should silently swallow the error. I'd raise `Liquid::ArgumentError`.